Tests/DatadogTests/Datadog/Tracing/Utils/MethodSwizzlerTests.swift

```diff
@@ -0,0 +1,163 @@
+            return originalRetVal + "...3rd party swizzled"
+        }
+        let newImp = imp_implementationWithBlock(newBlockImp)
+        method_setImplementation(method, newImp)
+    }
+
+    deinit {
+        let method: Method = class_getInstanceMethod(targetClass, selector)!
+        method_setImplementation(method, self.originalIMP!)
+    }
+}
+
+class MethodSwizzlerTests: XCTestCase {
+    private let swizzler = MethodSwizzler.shared
+
+    func testSimpleSwizzling() {
+        typealias TypedIMP = @convention(c) (AnyObject, Selector) -> String
+        typealias TypedBlockIMP = @convention(block) (AnyObject) -> String
+
+        let sel = #selector(BaseClass.methodToSwizzle as (BaseClass) -> () -> String)
+        let swizzledReturnValue = "this is swizzled imp"
+        let newBlockImp: TypedBlockIMP = { impSelf -> String in
+            return swizzledReturnValue
+        }
+        let newImp = imp_implementationWithBlock(newBlockImp)
+
+        let obj = BaseClass()
+
+        XCTAssertNoThrow(try swizzler.swizzle(selector: sel, in: BaseClass.self, with: newImp))
+        XCTAssertEqual(obj.perform(sel)?.takeUnretainedValue() as? String, swizzledReturnValue)
+    }
+
+    func testUnswizzling() {
+        typealias TypedIMP = @convention(c) (AnyObject, Selector) -> String
+        typealias TypedBlockIMP = @convention(block) (AnyObject) -> String
+
+        let sel = #selector(BaseClass.methodToSwizzle as (BaseClass) -> () -> String)
+        let swizzledReturnValue = "this is swizzled imp"
+        let newBlockImp: TypedBlockIMP = { impSelf -> String in
+            return swizzledReturnValue
+        }
+        let newImp = imp_implementationWithBlock(newBlockImp)
+
+        try! swizzler.swizzle(selector: sel, in: BaseClass.self, with: newImp)
+
+        let obj = BaseClass()
+
+        XCTAssertNoThrow(try swizzler.unswizzle(selector: sel, in: BaseClass.self))
+        XCTAssertEqual(obj.perform(sel)?.takeUnretainedValue() as? String, BaseClass.returnValue)
+    }
+
+    func testSwizzleSuperclassMethod() {
+        typealias TypedIMP = @convention(c) (AnyObject, Selector) -> String
+        typealias TypedBlockIMP = @convention(block) (AnyObject) -> String
+
+        let klass: AnyClass = EmptySubclass.self
+        let sel = #selector(EmptySubclass.methodToSwizzle as (EmptySubclass) -> () -> String)
+        let swizzledReturnValue = "this is swizzled imp"
+        let newBlockImp: TypedBlockIMP = { impSelf -> String in
+            return swizzledReturnValue
+        }
+        let newImp = imp_implementationWithBlock(newBlockImp)
+
+        do {
+            try swizzler.swizzle(selector: sel, in: klass, with: newImp)
+        } catch SwizzlingError.methodNotFound(let unfoundSelector, let searchedClassName) {
+            XCTAssertEqual(unfoundSelector, NSStringFromSelector(sel))
+            XCTAssertEqual(searchedClassName, NSStringFromClass(klass))
+        } catch {
+            XCTAssertNil(error)
+        }
```

Comment:
I don't understand this assertions 🤔. There should be only one valid control flow for asserting, but now it looks like this test is valid in two scenarios. What are we testing in this test and why we don't assert on `swizzledReturnValue`? Maybe we can also rename this test to `given/when/then` so we increases its readability.

If the goal is to test the error, we should use `XCTAssertThrowsError`:
```swift
XCTAssertThrowsError(try swizzler.swizzle(selector: sel, in: klass, with: newImp)) { error in
    guard case SwizzlingError.methodNotFound(let unfoundSelector, let searchedClassName) = error else {
        XCTFail("Expected `SwizzlingError.methodNotFound` but got \(error)")
        return
    }
    XCTAssertEqual(unfoundSelector, NSStringFromSelector(sel))
    XCTAssertEqual(searchedClassName, NSStringFromClass(klass))
}
```

Also, if the `swizzledReturnValue` is not important for this test, we should use `.mockAny()` to communicate that this can be any string:
```swift
let newBlockImp: TypedBlockIMP = { impSelf -> String in
    return .mockAny()
}
```